### PR TITLE
Structured stream udf 2.0

### PR DESCRIPTION
**Pull request by jenniew**
## What changes were proposed in this pull request?

Add structured stream with udf example, using file(parquet) as data source for streaming.

## How was this patch tested?

Run this example according to README

### CODE REVIEW

File: dl/src/main/scala/com/intel/analytics/bigdl/utils/DLClassifier.scala [hunk 1]
```diff
@@ -52,7 +52,7 @@ class DLClassifier[@specialized(Float, Double) T: ClassTag]
    */
   def setOutputCol(outputColName: String): this.type = set(outputCol, outputColName)
 
-  override def validateParams(): Unit = {
+  def validateParameters(): Unit = {
```
Comment (jason-dai) [inline dl/src/main/scala/com/intel/analytics/bigdl/utils/DLClassifier.scala]: why the change?

Comment (jenniew) [inline dl/src/main/scala/com/intel/analytics/bigdl/utils/DLClassifier.scala]: In spark mllib 2.1, trait Params does not have validateParams() any more. Or we may keep the DLClassifier code, and create Params wrapper in spark-version?

File: dl/src/main/scala/com/intel/analytics/bigdl/example/structuredStreamUdf/TextProducerParquet.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+              .save(param.destFolder)
+            count = 0
+            send_count += 1
+            Thread.sleep(interval*1000)
+          }
+
+        } catch {
+          case e: Exception => println(e)
+        }
+      }
```
Comment (yangw1234) [inline dl/src/main/scala/com/intel/analytics/bigdl/example/structuredStreamUdf/TextProducerParquet.scala]: Maybe we could use the `array.grouped` and `foreach` to simplify the while loop a little bit? Like
```
val batches = data.grouped(batchsize)
batches.foreach{ batch => ...}
```

**Comment by jenniew** [comment]
This PR is merged to PR #660
